underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java

```diff
@@ -421,30 +422,39 @@ protected ObjectListingChunk getObjectListingChunk(String key, boolean recursive
   // Get next chunk of listing result.
   private ListObjectsV2Result getObjectListingChunk(ListObjectsV2Request request)
       throws IOException {
-    ListObjectsV2Result result;
-    try {
-      // Query S3 for the next batch of objects.
-      result = mClient.listObjectsV2(request);
-      // Advance the request continuation token to the next set of objects.
-      request.setContinuationToken(result.getNextContinuationToken());
-    } catch (AmazonClientException e) {
-      throw new IOException(e);
+    // S3A write-then-list workload may has problems due to its eventual consistency issue
+    RetryPolicy retryPolicy = getRetryPolicy();
+    IOException thrownException = null;
+    while (retryPolicy.attempt()) {
+      try {
+        // Query S3 for the next batch of objects.
+        ListObjectsV2Result result = mClient.listObjectsV2(request);
+        // Advance the request continuation token to the next set of objects.
+        request.setContinuationToken(result.getNextContinuationToken());
+        return result;
+      } catch (AmazonClientException e) {
+        thrownException = new IOException(e);
```

Comment:
can we add debug logging here so that we can see retries in debug logs?

Reply:
added the debug logging, thanks